% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1.0in]{geometry}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{setspace}
\doublespacing
\usepackage[left]{lineno}
\linenumbers

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\hypertarget{resources-but-not-soil-particle-size-influence-ammonia-oxidizing-communities}{%
\section{Resources but not soil particle size influence ammonia
oxidizing
communities}\label{resources-but-not-soil-particle-size-influence-ammonia-oxidizing-communities}}

\vspace{10mm}

Nejc Stopnisek\({^\text{1}}\)\({^\dagger}\), Michelle
Quigley\({^\text{2}}\)\({^\dagger}\), Dan Chitwood\({^\text{2,3}}\) and
Ashley Shade\({^\text{1,4,5,6}}\)\({^\ddagger}\)

\vspace{20mm}

\({^\text{1}}\)Department of Microbiology \& Molecular Genetics,
Michigan State University, East Lansing, MI 48823, USA

\({^\text{2}}\)Department of Horticulture, Michigan State University,
East Lansing, MI 48823, USA

\({^\text{3}}\) Department of Computational Mathematics, Science and
Engineering, Michigan State University, East Lansing, MI 48823, USA

\({^\text{4}}\) Department for Plant, Soil and Microbial Sciences,
Michigan State University, East Lansing, MI 48823, USA

\({^\text{5}}\) Plant Resilience Institute, Michigan State University,
East Lansing, MI 48823, USA

\({^\text{6}}\) Great Lakes Bioenergy Research Center, Michigan State
University, East Lansing, MI 48823, USA

\({^\dagger}\) Contributed equally

\({^\ddagger}\) Corresponding author:
\href{mailto:shadeash@msu.edu}{shadeash@msu.edu}

\vspace{40mm}

\textbf{Key words}: Soil particles, microbiome, AOA, AOB, X-Ray CT,
ammonia oxidation

\newpage

\hypertarget{abstract}{%
\subsection{Abstract}\label{abstract}}

\newpage

\hypertarget{introduction}{%
\subsection{Introduction}\label{introduction}}

Ammonia oxidation first step in nitrification carried out by ammonia
oxidizing archaea (AOA) and ammonia oxidizing bacteria (AOB). Recently,
a group of bacteria (i.e.~comammox) has been described, known to perform
complete nitrification. While little is known about the distribution of
comammox organisms in the soil, we have a large number of studies
investigating the environmental and evolutionary preferences of AOA and
AOB in the soil environment. AOA have been shown to typically outnumber
the abundance of AOB in natural soils, however this is not true for
managed soils where nitrogen sources are usually added. AOA and AOB
commonly compete for same resources, however research has shown that AOA
activity is increased under low ammonium concentrations but reverse is
tru for AOB. Additionally, ammonium derived through mineralization is
thought to be the major source for AOA but organic ammonium for AOB.
Along with the ammonium resource distribution, pH seems to play an
important role in shaping the abundance and activity of AOA and AOB in
soil. Soil particles of different sizes are known to have different
chemical composition. Carbon and nitrogen content is thought to increase
with \_\_\_\_ and pH drops with size. Here mention the importance of
clay and water content.

We hypothesized that the co-occurrence of AOA and AOB in soil
environment is due to the soil chemical and size heterogeneity which
they preferentially and selectively occupy.

\newpage

\hypertarget{materials-and-methods}{%
\subsection{Materials and Methods}\label{materials-and-methods}}

\hypertarget{soil-collection-and-rhizotron-setup}{%
\subsubsection{Soil collection and rhizotron
setup}\label{soil-collection-and-rhizotron-setup}}

In September 2017, after harvesting the common beans, field soils were
collected from two location in Michigan, Montcalm Research Farm
(coordinates) and Saginaw Valley Extension and Research Center
(coordinates). Around 50 kg of soils were collected from the top 20 cm
and sieved through 4mm sieves on the location. Sieved soil was placed in
buckets and transported to the lab where it was stored at 4C until
rhizotron assembly. Back in the lab, rhizotrones were constructed, nine
for each sampled soil. The soil was gently added to the rhizotrones to
avoid over compactedness. Each rhizotron was divided by a barrier made
out of multilayered Miracloth into a larger compartment where up to 4
sprouted common bean seeds (\emph{Phaseolus vulgaris} cv. Eclipse) were
grown and the smaller compartment on the side of the rhizotrones where
no seeds were placed. Rhizotrones were covered with aluminum foil to
prevent light accessing to the root system. Plants were grown in growth
chambers (14h/10h and 26C/20C day/night cycle) and watered every 4 days
until plants reached V4 stage from when the plants were watered every 2
days.

\hypertarget{soil-sampling-and-sieving}{%
\subsubsection{Soil sampling and
sieving}\label{soil-sampling-and-sieving}}

When plants reached the senescence stage, soil was sampled by first
removing the aboveground biomass. Despite the barrier between the
compartments the roots penetrated it and grew into the seedless
compartment. Thus, only the large compartment fully grown with roots was
collected and is referred to as rhizosphere from now on. Soils were

\hypertarget{dna-isolations-pcr-and-sequencing}{%
\subsubsection{DNA isolations, PCR and
sequencing}\label{dna-isolations-pcr-and-sequencing}}

Using DNeasy PowerSoil DNA isolation kit (QIAGEN) to isolate DNA from
sieved soil fractions. 3 samples of 15 g were taken from each pooled
soil size fraction and these were treated as biological replicates
(totally 42 samples). From each of the replicate triplicate DNA
isolations were performed. each eluted in 25 ul buffer. The isolated DNA
from the same sample (n=3) was pooled, checked on agarose gels using gel
electrophoresis and quantified by Qubit using the dsDNA BR Assay kit.
For 16S rRNA amplicon sequencing, primer pair 515f and 806r was used
(Carporaso et al.) and sequencing was done using Illumina MiSeq v250
kit, yielding 250bp paired end reads. Amplification, library preparation
and sequencing was done at the Michigan State University Genomics Core
Facility. For ammonia oxidizing communities we first performed end point
PCR targeting amoA gene using specific primers for AOA, AOB and
comammox. While PCR counting AOA and AOB specific amoA primers were able
to produce amplicons in all soil size fractions, there was no
amplification of comammox amoA genes in these soils regardless of the
soil fraction. Thus, we focused on the AOA and AOB communities only. To
prepare amoA genes for sequencing, we first amplified them using primers
containing CS overhangs. 50 ul PCR mixtures were prepared for AOA and
AOB using primers (final concentration 600 nM), Pfu polymerase (Thermo
Scientific, USA) (1.5U) and Pfu Polymerase buffer (1x). The PCR
protocol.

\hypertarget{determining-the-soil-particle-surface-area}{%
\subsubsection{Determining the soil particle surface
area}\label{determining-the-soil-particle-surface-area}}

\hypertarget{amplicon-sequence-analysis}{%
\subsubsection{Amplicon sequence
analysis}\label{amplicon-sequence-analysis}}

\hypertarget{statistical-analysis}{%
\subsubsection{Statistical analysis}\label{statistical-analysis}}

\newpage

\hypertarget{results}{%
\subsection{Results}\label{results}}

We found that:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Microbial richness is influenced by the soil particle size.\\
\item
  Chemical properties of soil particles are very different and are site
  dependent.
\item
  Ammonia oxidizing communities are influenced by the ammonium
  concentrations and not soil article sizes.
\item
  Richness of AOA\textgreater AOB.
\item
  Soil particle surface area.
\end{enumerate}

Additional results:

\begin{itemize}
\tightlist
\item
  no detection of comammox in the system
\item
  the proportion of sieved fractions is (as \% of the whole soil)
  different between the two soils
\end{itemize}

\newpage

\hypertarget{discussion}{%
\subsection{Discussion}\label{discussion}}

\newpage

\hypertarget{acknowledgments}{%
\subsection{Acknowledgments}\label{acknowledgments}}

\hypertarget{references}{%
\subsection{References}\label{references}}

\newpage

\end{document}
